File: mobile/blender-buddy/core.bb/Models/CalculationResult.cs
using System.Collections.Generic;

namespace core.bb.Models
{
    public sealed class CalculationResult
    {
        public CalculationResult()
        {
            FillSpecs = new TankInfo();
            Warnings = new string[0];   
        }

        public MeasureMode System { get; set; }

        public decimal MaxDepth { get; set; }

        public decimal Po214Depth { get; set; }

        public decimal Po216Depth { get; set; }

        public decimal HypoxicDepth { get; set; }

        public decimal TopOffGas { get; set; }

        public TopOffGas TopOffGasType { get; set; }

        public TankInfo FillSpecs { get; set; }

        public IEnumerable<string> Warnings { get; set; }
    }
}

File: cordova/Blender-Buddy/scripts/Models/CalculationResult.ts
import { TankInfo } from './TankInfo';

export class CalculationResult {
    system: MeasureMode;
    topOffGasType: TopOffGas;
    fillSpecs: TankInfo;
    maxDepth: number;
    pO214Depth: number;
    pO216Depth: number;
    hypoxicDepth: number;
    topOffGas: number;
    warnings: string[];

    constructor() {
        this.system = MeasureMode.Imperial;
        this.topOffGasType = TopOffGas.Air;
        this.fillSpecs = new TankInfo();
    }
}